planetary_computer/sas.py

```diff
@@ -73,13 +80,41 @@ def sign(obj: Any) -> Any:
 
 
 @sign.register(str)
+def sign_string(url: str) -> str:
+    """Sign a URL or VRT-like string containing URLs with a Shared Access (SAS) Token
+
+    Signing with a SAS token allows read access to files in blob storage.
+
+    Args:
+        url (str): The HREF of the asset as a URL or a GDAL VRT
+
+            Single URLs can be found on a STAC Item's Asset ``href`` value. Only URLs to
+            assets in Azure Blob Storage are signed, other URLs are returned unmodified.
+
+            GDAL VRTs can combine many data sources into a single mosaic. A VRT can be
+            built quickly from the GDAL STACIT driver
+            https://gdal.org/drivers/raster/stacit.html. Each URL to Azure Blob Storage
+            within the VRT is signed.
+
+    Returns:
+        str: The signed HREF or VRT
+    """
+    if is_vrt_string(url):
```

Comment:
For the record, this doesn't seem to have a measurable effect on the timing of `planetary_computer.sign(url)`:

```python
# main, after signing a URL to get a cached token
In [5]: %timeit planetary_computer.sign(url)
12.2 µs ± 1.97 µs per loop (mean ± std. dev. of 7 runs, 100000 loops each)

# HEAD
In [4]: %timeit planetary_computer.sign(url)
11.3 µs ± 1.7 µs per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```